shell/components/form/PodAffinity.vue

```diff
@@ -257,8 +257,9 @@ export default {
             podAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution.push(term);
           }
         } else if (term.weight) {
-          const neu = { podAffinityTerm: term, weight: term.weight || this.defaultWeight };
+          const neu = { podAffinityTerm: { ...term }, weight: term.weight || this.defaultWeight };
 
+          delete neu.podAffinityTerm.weight;
```

Comment:
See https://github.com/rancher/dashboard/pull/9011 - I fixed antiAffinity weight and missed affinity. I think this PR covers testing this change as I've updated the payload comparison data to reflect the correct weighted pod affinity term spec.